qsiprep/utils/grouping.py

```diff
@@ -90,13 +96,28 @@ def get_entity_groups(layout, subject_data, combine_all_dwis):
         Each list of DWI files is a group of scans that can be concatenated together.
     """
     all_dwis = subject_data['dwi']
-    dwi_groups = []
     if not combine_all_dwis:
         dwi_groups = [[dwi] for dwi in all_dwis]
-
-    else:
-        dwi_entities = {}
-        for f in all_dwis:
+        return dwi_groups
+
+    all_metadata = []
+    for f in all_dwis:
+        metadata = layout.get_file(f).get_metadata()
+        all_metadata.append(metadata)
+
+    grouping_method = 'entities'  # default is to group by entities
+    if any('MultipartID' in metadata for metadata in all_metadata):
+        grouping_method = 'metadata'
+
+    dwi_entities = {}
+    grouping_metadata = {}
+    for i_file, f in enumerate(all_dwis):
+        if grouping_method == 'metadata':
+            # One MultipartID in any DWI metadata file means we use MultipartID to group
+            # If a DWI has no MultipartID, it will be placed in a group by itself
```

Comment:
Do we want to do this or do we want to group all those without multipart IDs? I think your approach is the right one here - if they use multipart IDs at all, it means that the curator has thought about what should be grouped together and has not specified a group for dwis with no multipart ID.  On the other hand, this is a different behavior than the default for empty multipart IDs. I think this is fine though

Reply:
I think it makes more sense to assume that only specifying MultipartIDs for some files means that the ones without that field should be considered independently, but it _is_ a change from the current behavior. The good thing is that the behavior will stay the same as long as no files have MultipartID.

I see it as somewhat analogous to IntendedFor. If you have IntendedFor pointing from the field maps to some of the DWI files, but other DWIs don't have anything intended for them, you should assume that the field maps shouldn't be applied to them.

Reply:
Great point. I agree